src/extension/platform-manager.js

```diff
@@ -0,0 +1,155 @@
+import url from 'url';
+import { exec } from 'mz/child_process';
+import _ from 'lodash';
+import { join } from 'path';
+import rmrf from 'rmfr';
+import ip from 'ip';
+import tmp from 'tmp-promise';
+import * as tunnel from '../extension/tunnel';
+import { spinify } from '../extension/spinner';
+import { startPackager } from './react-native';
+import { mobileEnvPath } from '../clients/cli-paths';
+import * as platform from './platform';
+import { readJsonFile, writeJsonFile } from './data';
+import { printMobilizerQR } from '../commands/qr-generator';
+import { getAppManager, getLegacyServiceClient } from '../clients/clients-factory';
+
+async function getAppDir(appId) {
+  return join(await mobileEnvPath(), appId.toString());
+}
+
+async function getCurrentApplicationState(appId, buildConfig) {
+  const publishingProperties = await (await getLegacyServiceClient()).getPublishingProperties(appId);
+  const installations = await (await getAppManager(appId)).getInstallations();
+  const installationsUrls = installations.map(inst => inst.location.app.package);
+  const platformVersion = await platform.getPlatformVersion(appId);
```

Comment:
this isn't good. You are chaining this async code instead of "parallel" running.
Use Promise.props for this case